### PR TITLE
Add Go 1.22 and 1.23, rm old versions, bump CI actions, bump golangci-linter

### PR DESCRIPTION
**Pull request by kolyshkin**
Mostly a CI refresh. See commits for details.

Closes: #769

### CODE REVIEW

**Review by tianon — APPROVED** [review]

**Comment by kolyshkin** [comment]
@opencontainers/runtime-tools-maintainers PTAL

File: .github/workflows/test.yml
Comment (tianon) [inline .github/workflows/test.yml:47]: We should still include 1.19 here for go.mod, right? Or do something clever with the `go-version-file:` support in setup-go?

Comment (kolyshkin) [reply .github/workflows/test.yml:47]: I'm not sure it makes sense to test on older Go releases, unless we explicitly want to maintain compatibility with older Go versions (and for now I don't know a reason to do that).

go version in `go.mod` will raise naturally once we'll start using newer features.

Comment (kolyshkin) [reply .github/workflows/test.yml:47]: Besides, we're switching to go 1.21 as a minimum in #777

**Comment by kolyshkin** [comment]
@opencontainers/runtime-tools-maintainers PTAL 🙏🏻

File: .github/workflows/test.yml
Comment (thaJeztah) [reply .github/workflows/test.yml:47]: Would it be ok to use;

- `min` version (go1.19 or go1.21 after #777)
- `latest`
- `latest - 1`

The last 2 being the currently supported versions, and the `min` version to verify that the minimum version specified in `go.mod` isn't lying.

Comment (kolyshkin) [reply .github/workflows/test.yml]: Updated; PTAL @thaJeztah

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by thaJeztah** [comment]
Oh, interesting; I have a green check here, but I'm not in the maintainers list for this repo. Wondering if some permissions were set too relaxed?

**Comment by kolyshkin** [comment]
> Oh, interesting; I have a green check here, but I'm not in the maintainers list for this repo. Wondering if some permissions were set too relaxed?

It looks like members of both runtime-tools-maintainers and runtime-spec-maintainers are admins here.

**Comment by kolyshkin** [comment]
@tianon we need a "still lgtm" from you here 🧔🏻

**Comment by kolyshkin** [comment]
@opencontainers/runtime-tools-maintainers PTAL 🙏🏻

**Review by tianon — APPROVED** [review]
Doh, sorry! 

To misquote one of my favorite people, "yep yep, looks like changes"

File: .github/workflows/test.yml
Comment (tianon) [reply .github/workflows/test.yml]: GitHub requires explicitly resolving conversations before merge; that's absolutely bonkers 🙃

**Pull request merged**
